app/move/controllers/create/document.js

```diff
@@ -1,24 +1,46 @@
-const formidable = require('formidable')
-const { get } = require('lodash')
+const multer = require('multer')
+const { reject } = require('lodash')
 
-const documentService = require('../../../../common/services/document')
-const moveService = require('../../../../common/services/move')
 const CreateBaseController = require('./base')
-const { API, FEATURE_FLAGS } = require('../../../../config')
+const documentService = require('../../../../common/services/document')
+const { FILE_UPLOADS, FEATURE_FLAGS } = require('../../../../config')
+
+const upload = multer({
+  dest: FILE_UPLOADS.UPLOAD_DIR,
+  limits: {
+    fileSize: FILE_UPLOADS.MAX_FILE_SIZE,
+  },
+})
+
+// TODO:
+// - bug with simulteaneous xhr uploads
 
 class DocumentUploadController extends CreateBaseController {
+  configure(req, res, next) {
+    req.form.options.fields.documents.xhrUrl = req.originalUrl
+    super.configure(req, res, next)
+  }
+
+  middlewareSetup() {
+    super.middlewareSetup()
+    this.use(upload.array('documents'))
+    // must be called after multer to correctly parse the form body
+    this.use(this.setNextStep)
+  }
+
   middlewareChecks() {
-    this.use(this.checkFeatureEnabled(FEATURE_FLAGS.DOCUMENTS))
-    this.use(this.parseMultipartForm)
+    this.use(this.isEnabled(FEATURE_FLAGS.DOCUMENTS))
     super.middlewareChecks()
   }
 
-  middlewareLocals() {
-    super.middlewareLocals()
-    this.use(this.populateDocumentUpload)
+  setNextStep(req, res, next) {
```

Comment:
Could this be let to HOF to handle, as a fork?

Reply:
The problem with doing it in the step config is that we don't have access to `req`. This is trying to say that if a file was uploaded or deleted with the form submission, redirect back to current URL rather than moving to the next step.

Maybe renaming this middleware might help? Something like `isUploadFormSubmission`?